### DISCUSSION TITLE
English Translate

### DISCUSSION THREAD

**Discussion by kyle-mckay** [Chimildic/goofy#73] [Беседка]
> ### Google Translate Русский
> Привет! Прежде всего, приносим свои извинения за грамматические ошибки или публикацию в неправильной категории. Английский - мой единственный язык, поэтому я просто запускаю все через Google Translate.
> 
> Я решил, что проведу обсуждение для любых англоязычных пользователей, желающих начать использовать Goofy, но русская документация и примечания в этом проекте могут показаться довольно пугающими, особенно для пользователей, практически не имеющих опыта программирования. Я решил начать анализировать код и любую официальную документацию и сделать ее более удобной для читателя. Очевидно, что это непростая задача без заявленных своевременных обещаний или чего-либо подобного, но я подумал, что начну с предоставления пользователям копии файла library.gs с ошибками консоли и комментариями, переведенными для удобства. Если это приведет к большему, пусть будет так, если нет, я надеюсь, по крайней мере, это поможет! До сих пор я запускал все свои функции вручную без каких-либо ошибок, поэтому я считаю, что он был преобразован без проблем.
> 
> Library.gs - Комментарии на английском языке / Консоль: [https://pastebin.com/fSrbbZuY](https://pastebin.com/fSrbbZuY) - Не уверен, смогу ли я опубликовать его здесь, поскольку я вообще не использую github, поэтому я бросил его на pastebin.
> 

### English
Hello! First of all apologies for any grammatical errors or posting in the wrong category. English is my only language so I am just running everything through Google Translate.

I figured I would put down a discussion for any English users wanting to start using Goofy but the Russian documentation and notes within this project can make it seem rather intimidating, especially for users with little to no coding experience. I've decided to start sifting through the code and any official documentation and making it a bit more reader friendly. Obviously this is a daunting task with no stated promises on time or anything along those lines, but I thought I would start by giving users a copy of the library.gs file with console errors and comments translated for convenience. If this leads to more, so be it, if not I hope at least this helps! So far I have ran all of my functions manually without any errors so I am of the belief that it has been converted without issue.

Library.gs - English Comments/Console: [https://pastebin.com/fSrbbZuY](https://pastebin.com/fSrbbZuY) - Not sure if I could post it in here as I do not use github much at all, so I threw it on pastebin.

**Comment by Chimildic** [comment]
Hello! You can translate `.md` files from [docs](https://github.com/Chimildic/goofy/tree/main/docs) and then I will add them to the [site](https://chimildic.github.io/goofy/). Files can be attached to message (bottom of edit form).

If you have trouble with errors or description, I can try to help.

**Reply by kyle-mckay** [reply]
Awesome, thanks for pointing out where the docs are stored! I do shift work for multiple weeks at a time, so when I get my time off at home I'm sure I'll spend some of it taking a look at these. 

Thanks for all the hard work you've put into Goofy, it is an incredibly useful tool! :)

**Reply by Chimildic** [reply]
In order to download files, click `code` -> `download zip` in the [main page](https://github.com/Chimildic/goofy). And then go to folder `docs`. 
The documentation is written in `markdown` language. Editing source files allows you to save hyperlinks, tables, pictures and so on.

![image](https://user-images.githubusercontent.com/40138097/117571644-1fed4500-b0d8-11eb-9e1c-ce5718bda878.png)

**Reply by kyle-mckay** [reply]
Thanks again for the information. I've gone ahead and made a fork of the repository into my account and started a branch for the [english translations](https://github.com/deadlypants0111/goofy/tree/English-Translation). For the time being I'm just going to push raw translations from in Google during my spare time until I have the time to review and adjust any grammatical changes back at home.
![branch](https://user-images.githubusercontent.com/83886427/117578402-65474d80-b0ab-11eb-8a0a-d923a571e3c6.png)
.

**Reply by kyle-mckay** [reply]
As the original instructions bring reference to them, I have decided to also work on images. Instead of replacing your images I will just add a trail to the filename. If it doesn't contain that trail there's no change in which file is being referenced from the docs folder. Trail is `_eng`
![2021-05-09 17_24_03-Greenshot](https://user-images.githubusercontent.com/83886427/117590303-07872580-b0ec-11eb-9bb5-d11f911aedca.png)

**Reply by kyle-mckay** [reply]
@Chimildic  Is there anything I can do while translating these that would make the process easier for you, or is just making a direct conversion/update to the documents for English enough? Figured I would ask in advance before I get ahead of myself!

> Могу ли я сделать что-нибудь при их переводе, что упростило бы вам процесс, или я просто делаю прямое преобразование / обновление документов для английского языка? Подумал, что я спрошу заранее, прежде чем забегу вперед!

**Reply by Chimildic** [reply]
You only need to save the `markdown` format. Such as link to pictures, quotes, blocks of code and so on. I mean the special symbols like `[] () | > - !` and other.
